vpr/src/route/route_timing.cpp

```diff
@@ -850,24 +886,39 @@ timing_driven_route_structs::~timing_driven_route_structs() {
                                      rt_node_of_sink);
 }
 
-void reduce_budgets_if_congested(route_budgets& budgeting_inf,
-                                 CBRR& connections_inf,
-                                 float slope,
+void reduce_budgets_if_congested(std::vector<ClusterNetId>& rerouted_nets,
+                                 route_budgets& budgeting_inf,
                                  int itry) {
-    auto& cluster_ctx = g_vpr_ctx.mutable_clustering();
-    if (budgeting_inf.if_set()) {
-        for (auto net_id : cluster_ctx.clb_nlist.nets()) {
-            if (should_route_net(net_id, connections_inf, false)) {
+    (void)rerouted_nets;
+    if (budgeting_inf.if_set() && itry > 20) {
```

Comment:
Why 20? Please change this magic numbers to named constants, with comments explaining the intention behind them.